src/serial_bridge_index.cpp

```diff
@@ -128,19 +126,18 @@ const char* serial_bridge::decompress(const char *buffer, size_t length) {
                 throw std::runtime_error("Decompression error");
         }
 
-        decompressedData.insert(decompressedData.end(), outBuffer, outBuffer + sizeof(outBuffer) - strm.avail_out);
-    } while (ret == Z_OK);
+        decompressedData.insert(decompressedData.end(), outBuffer, outBuffer + BUFFER_SIZE - strm.avail_out);
+    } while (strm.avail_out == 0);
```

Comment:
Do we also want to add `Z_STREAM_ERROR` to the error switch statement?
https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-inflate-1.html